subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -588,7 +589,7 @@ public ResolvedConfiguration getResolvedConfiguration() {
 
     private ResolveState resolveToStateOrLater(final InternalState requestedState) {
         assertIsResolvable();
-        warnIfConfigurationIsDeprecatedForResolving();
+        ensureValidConfigurationForDeclaration(name, resolutionAlternatives);
```

Comment:
I think this is the wrong error.  You're not allowed to resolve dependency buckets

Reply:
You're right, I saw the same upgrade guide section and replacement the error with the same check, which isn't correct at all here.